app/src/main/java/io/neurolab/main/task/SerialForwardTask.java

```diff
@@ -0,0 +1,132 @@
+package io.neurolab.main.task;
+
+import java.net.DatagramSocket;
+
+import io.neurolab.main.NFBServer;
+import io.neurolab.main.network.SerialForwarder;
+import io.neurolab.model.Config;
+import io.neurolab.model.DefaultFFTData;
+import io.neurolab.model.Task;
+
+public class SerialForwardTask implements Task {
+
+    long currentTimestamp = -1;
+    long nextRenderTimestamp = -1;
+    private float defaultMin = -2f;
+    private float defaultMax = 3f;
+    private int messageInterval = 75;
+    private DatagramSocket socket = null;
+    private String address;
+    private String baudRate;
+    private NFBServer nfbServer;
+    private float minValues[];
+    private float maxValues[];
+    private float rangeValues[];
+    private DefaultFFTData fftData;
+    private int mode = 0;     // 0 = per channel per bin, 1 = per channel per bin
+    private int forwardMode = 0;
+    private SerialForwarder serialForwarder;
+    private boolean serialForwardMaskInterfaceVisible;
+    private boolean connected = false;
+    private Config config;
+    private int[] messages;
+    private float[] vals;
+    
+    public SerialForwardTask(NFBServer nfbServer) {
+        this.nfbServer = nfbServer;
+        this.config = nfbServer.getConfig();
+        this.serialForwarder = new SerialForwarder();
+        // TODO: Instantiate a new SerialForwardMaskInterface object.
+    }
+
+    public void setSerialForwardMaskInterfaceVisible(boolean serialForwardMaskInterfaceVisible) {
+        this.serialForwardMaskInterfaceVisible = serialForwardMaskInterfaceVisible;
+    }
+
+    public synchronized void setMinVal(int index, float value) {
+        minValues[index] = value;
+        rangeValues[index] = maxValues[index] - minValues[index];
+    }
+
+    public synchronized void setMaxVal(int index, float value) {
+        maxValues[index] = value;
+        rangeValues[index] = maxValues[index] - minValues[index];
+    }
+
+    public synchronized void setCurrentFeedbackValues() {
+
```

Comment:
@suresh-gandhi Avoid writing empty public methods. Either comment the future motive or perform some action in it.